### ISSUE TITLE
Feature Request: graph order should not be "random" (alphabetically)

### ISSUE THREAD

**Issue by daganesh** [tensorflow/tensorboard#865]
Today the sections and the graphs are ordered alphabetically only (unless I miss something).
I would like to control this order as it has a meaning to see adjacent graphs and sections

thanks

**Comment by nfelt** [comment]
I assume by graph you mean the line charts on the "Scalars" dashboard?  (Not the "graph" plugin itself.)

You can customize the sections somewhat using the recently introduced Custom Scalars dashboard:
https://github.com/tensorflow/tensorboard/tree/master/tensorboard/plugins/custom_scalar

Since that exists, I'm not sure it makes sense to also support customization of just the section ordering in the scalars dashboard.  It might be possible to offer other predefined sort orders (like "most charts first" or "sort by timestamp of first event" but I'm not sure if those would meet your needs.

Note that it's a bit of a hack, but since the tag names are what's used for grouping, if you can control the tag names you can ultimately control the default order of the charts as well, e.g. if you had tags "bar", "baz", "foo", and you wanted them to instead be in the order "foo", "bar", "baz" you could give them prefixes like "1_foo", "2_bar", "3_baz".

**Comment by jart** [comment]
Correct. TensorBoard sorts tags and most things using a smart [lexicographical order](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/components/vz_sorting/sorting.ts#L17) which gives the user a lot of flexibility to control layout.

**Comment by daganesh** [comment]
thanks for the reply.
Indeed I was talking about the Scalar line charts, thanks for correcting.
I know I can use the names to control the order, but, as nfelt said: " it's a bit of a hack", and a bit ugly :)
What I expected by default, is that the sections and charts would be displayed in the same order I call them inside my code.
I will have a look at the new custom_scalar. I was not aware of it.
thanks again guys

**Comment by nfelt** [comment]
Interesting, thanks for elaborating.  It might be possible to detect and store the order of the summary calls within the code, but even if it is possible I'm not sure it would be in scope in the near future.

We can leave the feature request open for now, I'll update the title to be a bit more specific.  In the near term yeah, I'd suggest looking at the custom scalars plugin.
